hack/generator/pkg/config/configuration.go

```diff
@@ -184,24 +184,77 @@ func (config *Configuration) initialize(configPath string) error {
 	return kerrors.NewAggregate(errs)
 }
 
-// ShouldExport tests for whether a given type should be exported as Go code
-// Returns a result indicating whether export should occur as well as a reason for logging
-func (config *Configuration) ShouldExport(typeName astmodel.TypeName) (result ShouldExportResult, because string) {
-	for _, f := range config.ExportFilters {
-		if f.AppliesToType(typeName) {
-			switch f.Action {
-			case ExportFilterExclude:
+type ExportFilterFunc func(astmodel.TypeName) (result ShouldExportResult, because string)
+
+func buildExportFilterFunc(f *ExportFilter, allTypes astmodel.Types) ExportFilterFunc {
+	switch f.Action {
+	case ExportFilterExclude:
+		return func(typeName astmodel.TypeName) (ShouldExportResult, string) {
+			if f.AppliesToType(typeName) {
 				return Skip, f.Because
-			case ExportFilterInclude:
+			}
+
+			return "", ""
+		}
+
+	case ExportFilterInclude:
+		return func(typeName astmodel.TypeName) (ShouldExportResult, string) {
+			if f.AppliesToType(typeName) {
 				return Export, f.Because
-			default:
-				panic(errors.Errorf("unknown exportfilter directive: %s", f.Action))
+			}
+
+			return "", ""
+		}
+
+	case ExportFilterIncludeTransitive:
+		applicableTypes := make(astmodel.TypeNameSet)
+		for tn := range allTypes {
+			if f.AppliesToType(tn) {
+				getAllReferencedTypes(allTypes, tn, applicableTypes)
 			}
 		}
+
+		return func(typeName astmodel.TypeName) (ShouldExportResult, string) {
+			if applicableTypes.Contains(typeName) {
+				return Export, f.Because
+			}
+
+			return "", ""
+		}
+
+	default:
+		panic(errors.Errorf("unknown exportfilter directive: %s", f.Action))
+	}
+}
+
+func getAllReferencedTypes(allTypes astmodel.Types, root astmodel.TypeName, output astmodel.TypeNameSet) {
```

Comment:
Maybe `collectAllReferencedTypes`? A `get` function that doesn't return anything seems weird.